# Copyright (C) 2007  Peter De Wachter <pdewacht@gmail.com>
# Distributed under the same license as the original software. See debian/copyright.

Index: bulletml-0.0.6/src/bulletmlrunnerimpl.cpp
===================================================================
--- bulletml-0.0.6.orig/src/bulletmlrunnerimpl.cpp	2007-08-28 11:43:23.000000000 +0000
+++ bulletml-0.0.6/src/bulletmlrunnerimpl.cpp	2007-08-28 11:43:25.000000000 +0000
@@ -467,7 +467,7 @@
 		double dirSpace2;
 		if (dirSpace1 > 0) dirSpace2 = dirSpace1 - 360;
 		else dirSpace2 = dirSpace1 + 360;
-		if (abs(dirSpace1) < abs(dirSpace2)) dirSpace = dirSpace1;
+		if (fabs(dirSpace1) < fabs(dirSpace2)) dirSpace = dirSpace1;
 		else dirSpace = dirSpace2;
 
 		auto_ptr_copy(changeDir_, new LinearFunc<int, double>
Index: bulletml-0.0.6/src/formula-variables.h
===================================================================
--- bulletml-0.0.6.orig/src/formula-variables.h	2007-08-28 11:41:01.000000000 +0000
+++ bulletml-0.0.6/src/formula-variables.h	2007-08-28 11:43:25.000000000 +0000
@@ -33,7 +33,7 @@
 template <typename Val_>
 class Param : public AbstractNumber<Val_> {
 public:
-	DECLSPEC explicit Param(int id) : id_(id) {}
+	DECLSPEC explicit Param(unsigned int id) : id_(id) {}
 	DECLSPEC virtual Val_ value() const {
 		if (Variables::parameters && id_ < Variables::parameters->size()) {
 			return (*Variables::parameters)[id_];
@@ -44,7 +44,7 @@
 	}
 
 private:
-	int id_;
+	unsigned int id_;
 };
 
 #endif // ! FORMULA_VARIABLE_H_
